File: kitty-transcoder/kitty-transcoder-api/src/main/java/icu/jiapeng/kitty/transcoder/api/ProgressVO.java
package icu.jiapeng.kitty.transcoder.api;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.util.List;

@Data
public class ProgressVO {

    @Schema(description = "任务ID", example = "task_123456")
    private String taskId;

    @Schema(description = "转码进度（0-100）", example = "50")
    private Integer progress;

    @Schema(description = "当前状态", example = "PROCESSING")
    private String status;

    @Schema(description = "处理时间（秒）", example = "15")
    private Integer processingTime;

    @Schema(description = "当前步骤名称", example = "转码")
    private String currentStep;

    @Schema(description = "总步骤数", example = "3")
    private Integer totalSteps;

    @Schema(description = "各步骤进度列表")
    private List<StepProgressItem> stepProgressList;
}

File: kitty-transcoder-go/internal/model/progress_vo.go
package model

// ProgressVO 与 Java ProgressVO 对齐，用于 SSE 与回调
type ProgressVO struct {
	TaskID           string             `json:"taskId"`
	Progress         int                `json:"progress"`
	Status           string             `json:"status"`
	ProcessingTime   int                `json:"processingTime,omitempty"`
	CurrentStep      string             `json:"currentStep,omitempty"`
	TotalSteps       int                `json:"totalSteps,omitempty"`
	StepProgressList []StepProgressItem `json:"stepProgressList,omitempty"`
}

// StepProgressItem 与 Java StepProgressItem 对齐
type StepProgressItem struct {
	StepID   int    `json:"stepId"`
	Type     string `json:"type,omitempty"`
	Name     string `json:"name,omitempty"`
	Depends  string `json:"depends,omitempty"`
	Status   string `json:"status,omitempty"`
	Progress int    `json:"progress,omitempty"`
}
